[lib]/dgs/styleManager/Default/styleSettings.txt
---------------------------------------------------------------
--------------------------WARNING------------------------------
---This File Is Default Style Settings, Don't Edit This File!
---If This File Is Broken,Update DGS To Recover!
---------------------------------------------------------------
---How to make your own style?
---See: http://wiki.mtasa.com/Dgs/StyleManager
---The syntax of this file is lua table.
---------------------------------------------------------------
---------------------------------------------------------------

---------------------------------------------------------------------------------------------------General
	sharedTexture = true,	--Optimization ( Set to true to share texture element with the same creation data to prevent creating duplicated texture elements)
	sharedFont = true,		--Optimization ( Set to true to share font element with the same creation data to prevent creating duplicated texture elements)
	disabledColor = true,
	disabledColorPercent = 0.8,
	systemFont="default",	--For custom, systemFont="font.ttf", or systemFont={"font.ttf",Size,isBold,Quality} . This path is relative to the style path
	changeOrder=true,	--Change render order when click dgs element
---------------------------------------------------------------------------------------------------Cursor
	cursor = {
		enableCustomCursor = false,
		arrow = {										--arrow cursor
			image = {"Images/cursor/CursorArrow.png"},	--Image
			rotation = 0,								--Rotation
			rotationCenter = {0,0},						--Rotation Center (Relative)
			offset = {0,0},								--Offset (Relative)
			scale = 1,									--Scale
		},
		sizing_ns = {				           			--N-S (up-down) sizing cursor
			image = {"Images/cursor/CursorResize.png"},	--Image
			rotation = 0,								--Rotation
			rotationCenter = {0,0},						--Rotation Center (Relative)
			offset = {-0.5,-0.5},						--Offset (Relative)
			scale = 1.2,								--Scale
		},
		sizing_ew ={				            		--E-W (left-right) sizing cursor
			image = {"Images/cursor/CursorResize.png"},	--Image
			rotation = 90,								--Rotation
			rotationCenter = {0,0},						--Rotation Center (Relative)
			offset = {-0.5,-0.5},						--Offset (Relative)
			scale = 1.2,								--Scale
		},
		sizing_nwse = {				           			--NW-SE diagonal sizing cursor
			image = {"Images/cursor/CursorResize.png"},	--Image
			rotation = -45,								--Rotation
			rotationCenter = {0,0},						--Rotation Center (Relative)
			offset = {-0.5,-0.5},						--Offset (Relative)
			scale = 1.2,								--Scale
		},
		sizing_nesw = {				           			--NE-SW diagonal sizing cursor
			image = {"Images/cursor/CursorResize.png"},	--Image
			rotation = 45,								--Rotation
			rotationCenter = {0,0},						--Rotation Center (Relative)
			offset = {-0.5,-0.5},						--Offset (Relative)
			scale = 1.2,								--Scale
		},
		move = {					        			--Move
			image = {"Images/cursor/CursorMove.png"},	--Image
			rotation = 0,								--Rotation
			rotationCenter = {0,0},						--Rotation Center (Relative)
			offset = {-0.5,-0.5},						--Offset (Relative)
			scale = 1.2,								--Scale
		},
		text = {					        			--I-Beam cursor
			image = {"Images/cursor/CursorText.png"},	--Image
			rotation = 0,								--Rotation
			rotationCenter = {0,0},						--Rotation Center (Relative)
			offset = {-0.5,-0.5},						--Offset (Relative)
			scale = 1,									--Scale
		},
		pointer = {                            			--Pointer
			image = {"Images/cursor/CursorPointer.png"},--Image
			rotation = 0,								--Rotation
			rotationCenter = {0,0},						--Rotation Center (Relative)
			offset = {0,0},								--Offset (Relative)
			scale = 1,									--Scale
		},
	},
---------------------------------------------------------------------------------------------------Unique
-----------------------------------------------------------------------------------DGS 3D Text
	text3D = {
		font = nil,	--Using systemFont ( and usage is the same as system font )
	},

---------------------------------------------------------------------------------------------------Unique
-----------------------------------------------------------------------------------DGS Button
	button = {
	---Color
		textColor = tocolor(255,255,255,255),	--Button text color
		color = {	--Button background color
			tocolor(85,90,100,220),			--Normal
			tocolor(90,160,230,220),		--Hover
			tocolor(60,110,180,220),		--Click
		},
	---Image
		image = {	--Button background color
			{},	--Normal
			{},	--Hover
			{},	--Click
		},
	---Values
		font = nil,	--Using systemFont ( and usage is the same as system font )
		textSize = {1,1},
	},

-----------------------------------------------------------------------------------DGS Check Box
	checkbox = {
	---Color
		textColor = tocolor(255,255,255,255),	--Text Color
		color_f = {	--Default image color when state is false ( unchecked )
			tocolor(255,255,255,255),	--Normal
			tocolor(255,255,255,255),	--Hover
			tocolor(180,180,180,255),	--Click
		},
		color_i = {	--Default image color when state is nil ( indeterminate )
			tocolor(255,255,255,255),	--Normal
			tocolor(255,255,255,255),	--Hover
			tocolor(180,180,180,255),	--Click
		},
		color_t = {	--Default image color when state is true ( checked )
			tocolor(255,255,255,255),	--Normal
			tocolor(255,255,255,255),	--Hover
			tocolor(180,180,180,255),	--Click
		},
	---Image
		image_f = {	--Default image when state is false ( unchecked )
			{"Images/checkbox/cb_f.png"},	--Normal
			{"Images/checkbox/cb_f_s.png"},	--Hover
			{"Images/checkbox/cb_f.png"},	--Click
		},
		image_i = {	--Default image when state is nil ( indeterminate )
			{"Images/checkbox/cb_i.png"},	--Normal
			{"Images/checkbox/cb_i_s.png"},	--Hover
			{"Images/checkbox/cb_i.png"},	--Click
		},
		image_t = {	--Default image when state is true ( checked )
			{"Images/checkbox/cb_t.png"},	--Normal
			{"Images/checkbox/cb_t_s.png"},	--Hover
			{"Images/checkbox/cb_t.png"},	--Click

		},
	---Values
		font = nil,	--Using systemFont ( and usage is the same as system font )
		textSize = {1,1},	--Text size
		textPadding = {
			2,		--Space
			false,	--Relative
		},
		buttonSize = {
			16,		--Button size
			false,	--Relative
		},
	},

-----------------------------------------------------------------------------------DGS Combo Box
	combobox = {
	---Color
		bgColor = false,	--Background color of the combo box
		textColor = tocolor(255,255,255,255),		--Text color
		color = {	--Background color of the arrow button
			tocolor(85,90,100,220),			--Normal
			tocolor(90,160,230,220),		--Hover
			tocolor(60,110,180,220),		--Select
		},
		itemColor = {	--Background color of items
			tocolor(30,32,35,255),			--Normal
			tocolor(80,85,90,255),			--Hover
			tocolor(70,140,210,255),		--Select
		},
		itemTextColor = tocolor(255,255,255,255),	--Item text color
		arrowColor = tocolor(255,255,255,255),
		arrowOutSideColor = tocolor(255,255,255,255),
	---Image
		arrow = {	--Arrow Image
			"Images/combobox/arrow.fx",	--File Path
			"shader",					--Texture type (Default is "image")
			{},							--Shader Settings
		},
		image = {
			{},	--Normal
			{},	--Hover
			{},	--Select
		},
		itemImage = {
			{},	--Normal
			{},	--Hover
			{},	--Select
		},
		bgImage = {},
	---Values
		font = nil,	--Using systemFont ( and usage is the same as system font )
		textSize = {1,1},
		itemTextSize = {1,1},
		itemHeight = 20,
		arrowSettings = {0.2,0.1,0.02},
		itemTextPadding = {5,5},
		textPadding = {5,5},
		scrollBarThick = 15,
		autoHideAfterSelected=true,
	},

-----------------------------------------------------------------------------------DGS Radio Button
	radiobutton = {
	---Color
		textColor = tocolor(255,255,255,255),	--Text Color
		color_f = {	--Default image color when state is false ( unselected )
			tocolor(255,255,255,255),	--Normal
			tocolor(255,255,255,255),	--Hover
			tocolor(180,180,180,255),	--Click
		},
		color_t = {	--Default image color when state is true ( selected )
			tocolor(255,255,255,255),	--Normal
			tocolor(255,255,255,255),	--Hover
			tocolor(180,180,180,255),	--Click
		},
	---Image
		image_f = {	--Default image when state is false ( unchecked )
			{"Images/radiobutton/rb_f.png"},		--Normal
			{"Images/radiobutton/rb_f_s.png"},	--Hover
			{"Images/radiobutton/rb_f.png"},		--Click
		},
		image_t = {	--Default image when state is true ( checked )
			{"Images/radiobutton/rb_t.png"},		--Normal
			{"Images/radiobutton/rb_t_s.png"},	--Hover
			{"Images/radiobutton/rb_t.png"},		--Click
		},
	---Values
		font = nil,	--Using systemFont ( and usage is the same as system font )
		textSize = {1,1},	--Text size
		textPadding = {
			2,		--Space
			false,	--Relative
		},
		buttonSize = {
			16,		--Button size
			false,	--Relative
		},
	},

-----------------------------------------------------------------------------------DGS Edit
	edit = {
	---Color
		bgColor = tocolor(85,90,100,255),		--Background color of edit
		bgColorBlur = nil,		--Background color of edit when blur, set to nil to use bgColor
		textColor = tocolor(255,255,255,255),			--Text color
		caretColor = tocolor(255,255,255,255),		--Caret color
		selectColor = tocolor(50,150,255,200),	--Select background color
		selectColorBlur = tocolor(140,140,140,200),	--Selected background color when blurred
		placeHolderColor = tocolor(200,200,200,255),		--Place holder text color
	---Image
		bgImage = {},	--Background image of edit
		bgImageBlur = {},	--Background image of edit when blur, set to empty or nil to use bgImage
	---Values
		font = nil,	--Using systemFont ( and usage is the same as system font )
		textSize = {1,1},	--Text size
		maskText = "*",
		padding = {2,2},
		caretStyle = 0,
		caretThick = 1.2,
		caretOffset = 0,
		caretHeight = 1,
		typingSound = nil,
		typingSoundVolume = 1,
		placeHolder = "",
		placeHolderColorcoded = false,
		placeHolderOffset = {0,0},
		placeHolderTextSize = nil,	---Use textSize
		placeHolderIgnoreRenderTarget = false,
	},

-----------------------------------------------------------------------------------DGS Grid List
	gridlist = {
	---Color
		bgColor = tocolor(30,32,35,225),		--Background color of grid list
		columnColor = tocolor(30,32,35,255),	--Background color of column
		columnTextColor = tocolor(255,255,255,255),	--Column text color
		rowColor = {	--Background color of row
			tocolor(30,32,35,255),			--Normal
			tocolor(54,57,64,255),			--Hover
			tocolor(70,140,210,255),		--Select
		},
		rowTextColor = tocolor(255,255,255,255),	--Row text color
	---Image
		bgImage = {},
		columnImage = {},
		rowImage = {
			{},
			{},
			{},
		},

	---Values
		font = nil,	--Using systemFont ( and usage is the same as system font )
		columnTextSize = {1,1},
		rowTextSize = {1,1},
		columnOffset = 10,
		columnHeight = 20,
		rowHeight = 15,
		sectionColumnOffset = -10,
		defaultColumnOffset = 0,
		backgroundOffset = -10,
		scrollBarThick = 15,
	},

-----------------------------------------------------------------------------------DGS Label
	label = {
	---Color
		textColor = tocolor(255,255,255,255),	--Label text color
	---Image

	---Values
		font = nil,	--Using systemFont ( and usage is the same as system font )
		textSize = {1,1},
	},

-----------------------------------------------------------------------------------DGS Memo
	memo = {
	---Color
		bgColor = tocolor(85,90,100,255),	--Background color of memo
		bgColorBlur = nil,		--Background color of memo when blur, set to nil to use bgColor
		textColor = tocolor(255,255,255,255),		--Text color
		caretColor = tocolor(255,255,255,255),	--Caret color
		selectColor = tocolor(50,150,255,200),	--Select background color
		selectColorBlur = tocolor(140,140,140,200),	--Selected background color when blurred
	---Image
		bgImage = {},	--Background image of memo
		bgImageBlur = {},	--Background image of memo when blur, set to empty or nil to use bgImage
	---Values
		font = nil,	--Using systemFont ( and usage is the same as system font )
		textSize = {1,1},	--Text size
		maskText = "*",
		caretStyle = 0,
		caretThick = 1.2,
		caretOffset = 3,
		caretHeight = 1,
		scrollBarThick = 15,
		selectVisible = true,
		typingSound = nil,
		typingSoundVolume = 1,
		padding = {2,2},
	},

-----------------------------------------------------------------------------------DGS Progress Bar
	progressbar = {
	---Color
		bgColor = tocolor(5,10,20,200),		--Background color of progress bar
		indicatorColor = tocolor(80,140,210,200),--Background color of progress bar indicator
	---Image
		bgImage = {},
		indicatorImage = {},
	---Values
		padding = {2,2},
	},

-----------------------------------------------------------------------------------DGS Scroll Bar
	scrollbar = {
	---Color
		cursorColor = {	--Cursor color
			tocolor(79,83,91,255),		--Normal
			tocolor(90,160,230,255),		--Hover
			tocolor(60,110,180,255),		--Click
		},
		arrowColor = {	--Arrow color
			tocolor(240,240,240,255),		--Normal
			tocolor(90,160,230,255),		--Hover
			tocolor(60,110,180,255),		--Click
		},
		troughColor = {	--Trough color
			tocolor(30,30,30,255),	--Part 1
			tocolor(30,30,30,255),	--Part 2
		},
	---Image
		arrowImage = {"Images/scrollbar/scrollbar_arrow.png"},		--Arrow image
		cursorImage = nil,											--Cursor image
		troughImage = nil,											--Trough image for vertical and horizontal
		troughImageHorizontal = nil,								--Trough image for horizontal only
		arrowBgColor = {
			tocolor(30,30,30,255),	--Normal
			tocolor(30,30,30,255),	--Hover
			tocolor(30,30,30,255),	--Click
		},
	---Values
		scrollArrow = true,
		cursorWidth = {1,true},
		troughWidth = false,	--False for using cursorWidth
		arrowWidth = false,	--False for using cursorWidth
		imageRotation = {
			{270,270,0},--{Horizontal}, (arrows,cursor,trough)
			{0,0,270},--{Vertical}, (arrows,cursor,trough)
		}
	},

-----------------------------------------------------------------------------------DGS Scroll Pane
	scrollpane = {
	---Color

	---Image

	---Values
		scrollBarThick = 15,
	},
	
-----------------------------------------------------------------------------------DGS Scale Pane
	scalepane = {
	---Color

	---Image

	---Values
		scrollBarThick = 15,
	},

-----------------------------------------------------------------------------------DGS Selector
	selector = {
	---Color
		itemTextColor = tocolor(255,255,255,255),	--Item Text Color
		selectorText = {"<",">"},
		selectorTextColor = {
			tocolor(255,255,255,255),		--Normal
			tocolor(90,160,230,220),		--Hover
			tocolor(60,110,180,220),		--Click
		},
		selectorImageLeft = {
			nil,		--Normal
			nil,		--Hover
			nil,		--Click
		},
		selectorImageColorLeft = {
			nil,		--Normal
			nil,		--Hover
			nil,		--Click
		},
		selectorImageRight = {
			nil,		--Normal
			nil,		--Hover
			nil,		--Click
		},
		selectorImageColorRight = {
			nil,		--Normal
			nil,		--Hover
			nil,		--Click
		},
	---Values
		font = nil,	--Using systemFont ( and usage is the same as system font )
		selectorTextSize = {1,1},
		itemTextSize = {1,1},

	},
-----------------------------------------------------------------------------------DGS Switch Buttoon
	switchbutton = {
	---Color
		textColorOn = tocolor(250,250,250,255),	--Text Color
		textColorOff = tocolor(250,250,250,255),	--Text Color
		colorOn = {	--Default image color when state is true (On)
			tocolor(90,160,230,255),	--Normal
			tocolor(90,160,230,255),	--Hover
			tocolor(90,160,230,255),	--Click
		},
		colorOff = {	--Default image color when state is false (Off)
			tocolor(60,60,60,255),	--Normal
			tocolor(60,60,60,255),	--Hover
			tocolor(60,60,60,255),	--Click
		},
		cursorColor = {	--Default image color when state is true ( selected )
			tocolor(220,220,220,255),	--Normal
			tocolor(240,240,240,255),	--Hover
			tocolor(200,200,200,255),	--Click
		},
	---Image
		imageOn = {	--Default image when state is false ( unchecked )
			{},		--Normal
			{},		--Hover
			{},		--Click
		},
		imageOff = {	--Default image when state is true ( checked )
			{},		--Normal
			{},	--Hover
			{},		--Click
		},
		cursorImage = {
			{},
			{},
			{},
		},
	---Values
		font = nil,	--Using systemFont ( and usage is the same as system font )
		textSize = {1,1},
		textOffset = {0.25,true},
		troughWidth = {1,true},
		cursorWidth = {1,true},
		cursorLength = {0.5,true},
	},

-----------------------------------------------------------------------------------DGS Tab Panel
	tabpanel = {
	---Color
		bgColor = tocolor(54,57,64,255),	--Background color of tab panel
		textColor = {						--If tab doesn't have color, use this one.
			tocolor(255,255,255,255),	--Normal
			tocolor(255,255,255,255),	--Hovered
			tocolor(255,255,255,255),	--Selected
		},
	---Image
		bgImage = {},
	---Values
		font = nil,	--Using systemFont ( and usage is the same as system font )
		tabPadding = {4,false},
		tabGapSize = {1,false},
		scrollSpeed = {10,false},
		tabHeight = 20,
	},

-----------------------------------------------------------------------------------DGS Tab
	tab = {
	---Color
		textColor = tocolor(255,255,255,255),		--Use Tab Panel's
		bgColor = nil,			--Background color of tab panel when selected(This will overwrite the background color of tab panel when selected). nil for using tab panel's bgColor
		tabColor = {	--Background color of tab
			tocolor(50,50,50,180),	--Normal
			tocolor(80,80,80,190),	--Hovered
			tocolor(54,57,64,255),	--Selected
		},
	---Image
		bgImage = {},
		tabImage = {
			{},
			{},
			{},
		},
	---Values
		font = nil,	--Using tab panel's font
		textSize = {1,1},
	},

-----------------------------------------------------------------------------------DGS Window
	window = {
	---Color
		textColor = tocolor(255,255,255,255),		--Window title text color
		titleColor = tocolor(40,40,40,240),			--Window title background color
		titleColorBlur = tocolor(20,20,20,240),		--Window title background color when blurred
		color = tocolor(30,32,35,235),				--Window background color
		closeButtonColor = {	--Window close button color
			tocolor(0,0,0,0),	--Normal
			tocolor(250,20,20,255),	--Hover
			tocolor(150,50,50,255),	--Click
		},
	---Image
		image = {},
		titleImage = {},
	---Values
		closeButton = true,
		font = nil,	--Using systemFont ( and usage is the same as system font )
		textSize = {1,1},
		titleHeight = 25,
		borderSize = 5,
		closeButtonText = "×",
	},


[skygfx]/skygfx/config.lua
TIMECYC = exports.timecyc
COR = exports.custom_coronas

-- SA Timecyc.dat mapping
T = {
    ambR =1 , ambG =2 , ambB =3,
    ambR_obj =4 , ambG_obj =5 , ambB_obj =6,
    dirR =7, dirG =8, dirB =9,
    skyTopR = 10, skyTopG =11, skyTopB =12,
    skyBotR = 13, skyBotG =14, skyBotB =15,
    sunCoreR = 16, sunCoreG =17, sunCoreB =18,
    sunCoronaR = 19, sunCoronaG =20, sunCoronaB =21,
    sunSz = 22, sprSz =23, sprBght =24,
    shad = 25, lightShad =26, poleShad =27,
    farClp = 28, fogSt =29, lightGnd =30,
    cloudR = 31, cloudG =32, cloudB =33,
    fluffyBotR =34, fluffyBotG =35, fluffyBotB =36,
    waterR = 37,waterG =38, waterB=39, waterA=40,
    postfx1A =41, postfx1R =42, postfx1G = 43, postfx1B = 44,
    postfx2A =45, postfx2R = 46, postfx2G =47, postfx2B = 48,
    cloudAlpha = 49, radiosityLimit = 50, waterFogAlpha = 51, dirMult = 52
}

textureListTable = { }

textureListTable.BuildingPSRemoveList = {
    "",	"unnamed", "fire*",                                    -- unnamed
    "basketball2","skybox_tex*",                               -- other
    "font*","radar*","sitem16","snipercrosshair",              -- hud
    "siterocket","cameracrosshair",                            -- hud
    "*shad*",                                                  -- shadows
    "coronastar","coronamoon","coronaringa",
    "coronaheadlightline",                                     -- coronas
    "lunar",                                                   -- moon
    "tx*",                                                     -- grass effect
    "cj_w_grad",                                               -- checkpoint texture
    "*cloud*",                                                 -- clouds
    "*smoke*",                                                 -- smoke
    "sphere_cj",                                               -- nitro heat haze mask
    "water*","newaterfal1_256",
    "boatwake*","splash_up","carsplash_*",
    "fist","*icon","headlight*",
    "unnamed","sphere","plaintarmac*",
    "vehiclegrunge256","?emap*","vehiclegeneric*",
    "gensplash"
}
textureListTable.VehiclePSApplyList = {
    "predator92body128", "monsterb92body256a", "monstera92body256a", "andromeda92wing","fcr90092body128",
    "hotknifebody128b", "hotknifebody128a", "rcbaron92texpage64", "rcgoblin92texpage128", "rcraider92texpage128", 
    "rctiger92body128","rhino92texpage256", "petrotr92interior128","artict1logos","rumpo92adverts256","dash92interior128",
    "coach92interior128","combinetexpage128","hotdog92body256",
    "raindance92body128", "cargobob92body256", "andromeda92body", "at400_92_256", "nevada92body256",
    "polmavbody128a" , "sparrow92body128" , "hunterbody8bit256a" , "seasparrow92floats64" , 
    "dodo92body8bit256" , "cropdustbody256", "beagle256", "hydrabody256", "rustler92body256", 
    "shamalbody256", "skimmer92body128", "stunt256", "maverick92body128", "leviathnbody8bit256",
    "vehiclegrunge256","?emap*"
}

SKYGFX = {
    ps2Modulate=true,
    dualPass=true,
    zwriteThreshold=128,
    -- postfx
    colorFilter = "PS2", -- values: "PS2", "PC", "Mobile", (needs colorcycle.dat), "None"
    blurLeft = 0.0004, -- Override PS2 color filter blur offset 
    blurTop = 0.0004, -- to disable blur set these to 0
    blurRight = 0.0004,
    blurBottom = 0.0004,
    doRadiosity=true,-- Enable or disable radiosity
    radiosityFilterPasses=2,
    radiosityRenderPasses=1,
    radiosityIntensity=35,
    radiosityIntensityLimit=0, -- use to override the intensity limit, 0 for use the value from timecyc (by default)
    usePCTimecyc = false,
    RSPIPE_PC_CustomBuilding_PipeID = true,
    fogDisabled = false,
    brightnessMul = 1,
    -- grass
    grassAddAmbient=false, --0x5DAEC8, need fuck the memory, not fully done.
    --grassFixPlacement=true, 0x5DADB7, need fuck the memory
    grassBackfaceCull=true,
    vehiclePipe= "PS2", -- values: "PS2", "PC", "Xbox", "Spec" (like PS2 but with specular lighting), "Neo" (like III/VC Xbox), "LCS", "VCS", "Mobile", "Env"

    -- special
    vehicleBigHeadLight = true, -- show vc/iii liked vehicle big headlight
    vehicleTrail = true, -- show vc/iii liked vehicle light trail
    vehicleTrailLength = 6, -- length of buffered frame
    vehicleTrailDrawDist = 20,
   
}
w, h = guiGetScreenSize()